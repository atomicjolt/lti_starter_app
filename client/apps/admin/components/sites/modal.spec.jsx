import React from 'react';
import TestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import Helper from '../../../../specs_support/helper';
import Modal from './modal';

describe('site modal', () => {
  let result;

  const url = 'bfcoder.com';

  const props = {
    isOpen: true,
    closeModal: () => {},
    site: {
      id: 1,
      url
    },
    createSite: () => {},
    updateSite: () => {},
    settings: {
      lti_key: '',
      canvas_callback_url: '',
    },
  };

  beforeEach(() => {
    result = TestUtils.renderIntoDocument(
      <Provider store={Helper.makeStore()}>
        <Modal {...props} />
      </Provider>
    );
  });

  it('uses the application url as the title', () => {
    const modal = TestUtils.findRenderedComponentWithType(result, Modal);
    expect(modal).not.toBeUndefined();
    // TODO we need better specs but reaching into the modal has turned out to be a pain.
    // const h2s = TestUtils.scryRenderedDOMComponentsWithTag(result, 'h2');
    // const h2 = _.find(h2s, { className: 'c-modal__title' });
    // expect(h2.textContent).toContain(name);
  });

});
