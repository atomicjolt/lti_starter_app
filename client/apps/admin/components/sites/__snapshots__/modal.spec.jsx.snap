// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site modal matches the snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="c-modal c-modal--site is-open"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Modal"
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="unused"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2
    className="c-modal__title"
  >
    Update
     Domain
  </h2>
  <h3
    className="c-modal__subtext"
  >
    <div>
      <div>
        Use the redirect URI: 
      </div>
      <div>
        <a
          className="c-modal__subtext-link"
          href="bfcoder.com/accounts/site_admin/developer_keys"
          rel="noopener noreferrer"
          target="_blank"
        >
          Get Canvas Developer Keys Here
        </a>
      </div>
    </div>
  </h3>
  <h3
    className="c-modal__subtext"
  >
    Url must begin with "https://"
  </h3>
  <SiteForm
    closeModal={[Function]}
    id={1}
    isUpdate={true}
    onChange={[Function]}
    setupSite={[Function]}
    url="bfcoder.com"
  />
</Modal>
`;
